nautobot/core/models/__init__.py

```diff
@@ -89,6 +90,13 @@ def present_in_database(self):
         """
         return not self._state.adding
 
+    @property
+    def is_contact_associable_model(self):
+        """
+        True if the record can be associated with a contact/team, False if it cannot.
+        """
+        return f"{self._meta.app_label}.{self._meta.model_name}" not in NON_CONTACT_ASSOCIABLE_MODELS
+
```

Comment:
Rather than a property with a hard-coded list, can we just set a flag on the model class or model Meta to this effect? I can imagine that there may be app models that will similarly want to opt out, for example.

Reply:
Epitome of over-engineering